### ISSUE TITLE
Adding CI with Sanitizers

### ISSUE THREAD

**Issue by ax3l** [openpmix/openpmix#1025]
Hi,

are you planning to add CI to the PMIx repo?

One build could include a build and run with `-fsanitize=address` in order to avoid introducing memory leaks and violations in updates.

**Comment by jjhursey** [comment]
PR #1265 is re-enabling the `-fsanitize=address` version in the Travis config:
 * https://github.com/openpmix/openpmix/blob/master/.travis.yml#L32-L36

We need community help on plugging the rest of the leaks to get it to pass.

**Comment by ax3l** [comment]
Little note: my PR is not re-enabling them, but enabling them for the first time ;) I just added them as comments when adding CI in the first place.

**Comment by rhc54** [comment]
@ax3l I finally got around to doing this in the code base, enabling the "picky" compiler options and adding the `sanitize=address` option. Even got it all to build and work on my Mac.

However, when trying to get it thru CI, I ran into problems with compilers that support that sanitize option, but appear to require the addition of another library (believe it might be -lasan?). I'm having trouble finding the correct autoconf code to locate that library.

Any suggestions?
